src/celestia/qt/qtdeepskybrowser.cpp

```diff
@@ -184,8 +184,13 @@ void populateDsoVector(std::vector<DeepSkyObject*>& dsos,
             return;
         }
 
-        if (DeepSkyObject* dso = dsodb.getDSO(index); filter(dso))
+        DeepSkyObject* dso = dsodb.getDSO(index);
+        if (filter(dso))
         {
+            std::string dson = dsodb.getDSOName(dso, true).c_str();
```

Comment:
```suggestion
            std::string dson = dsodb.getDSOName(dso, true);
```
`getDSOName` returns std::string, not need to convert it